src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -114,8 +132,16 @@ internal JsonClassInfo(Type type, JsonSerializerOptions options)
             }
         }
 
-        internal JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadStackFrame frame)
+        internal JsonPropertyInfo GetProperty(JsonSerializerOptions options, ReadOnlySpan<byte> propertyName, ref ReadStackFrame frame)
         {
+            // If we should compare with case-insensitive, normalize to an uppercase format since that is what is cached on the propertyInfo.
+            if (options.PropertyNameCaseInsensitive)
+            {
+                string utf16PropertyName = Encoding.UTF8.GetString(propertyName.ToArray());
```

Comment:
That is a lot of allocations.

Could we use the span, in-place to upper APIs and the Operation-status basic UTF8 encoder?
https://github.com/dotnet/corefx/blob/3ac254fa7b3081b94c45af2d780f49b5476f5614/src/System.Memory/ref/System.Memory.cs#L94

```C#
namespace System.Text.Unicode
{
    public static partial class Utf8
    {
        public static System.Buffers.OperationStatus FromUtf16(ReadOnlySpan<char> source, Span<byte> destination, out int charsRead, out int bytesWritten, bool replaceInvalidSequences = true, bool isFinalBlock = true) { throw null; }
        public static System.Buffers.OperationStatus ToUtf16(ReadOnlySpan<byte> source, Span<char> destination, out int bytesRead, out int charsWritten, bool replaceInvalidSequences = true, bool isFinalBlock = true) { throw null; }
    }
}
```

Reply:
This is going to be re-written when https://github.com/dotnet/corefx/issues/36939 is addressed - instead it will just use the existing utf16 property name with a case-insensitive dictionary.